treeherder/seta/jobtypes.py

```diff
@@ -0,0 +1,152 @@
+    try:
+        group = name.split('-')[0]
+    except:
+        group = ''
+    return group
+
+
+def _getgroupCode(tbplnames, name):
+    """
+    It's about get group code by given testname, like T-e10s for tp5o-e10s.
+    And there has some useless blank in testname like " mochitest-devtools-chrome-3"
+    to strip that blank off, so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['group']
+    except:
+        code = ''
+    return code
+
+
+def _getcode(tbplnames, name):
+    """
+    It's about get the code by given testname, it usually related with the group code
+    like '2' for web-platform-tests-2.   And there has some useless blank in testname
+    like " mochitest-devtools-chrome-3" to strip that blank off,
+    so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['code']
+    except:
+        code = ''
+    return code
+
+
+class Treecodes:
+    """This class contain all the mapping we need in SETA and make it works"""
+
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        self.jobtypes = []
+        self.jobnames = []
+
+        ManageJobPriorityTable().update_job_priority_table()
+
+        with open(get_runnable_jobs_path()) as data:
+            joblist = json.loads(data.read())['results']
+
+        for job in joblist:
+            # e.g. web-platform-tests-4
+            # e.g. Ubuntu VM 12.04 x64 mozilla-inbound opt test web-platform-tests-4 OR
+            #      desktop-test-linux64/opt-web-platform-tests-4
+            testtype, job_identifier = self._testtype_job_identifier(job)
+
+            # XXX: This has the danger of falling out of date
+            if not testtype or 'mulet' in job['platform']:
+                LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                continue
+
+            # XXX: This has the danger of falling out of date
+            for i in ('dep', 'nightly', 'non-unified', 'valgrind', 'build'):
+                if testtype.find(i) != -1:
+                    LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                    continue
+
+            try:
+                platform = job['platform']
+                buildtype = job['platform_option']
+                self.jobtypes.append([platform, buildtype, testtype])
+
+                # It's about get jobnames and both buildbot and taskcluster job has '?'
+                # when the job_group_symbol is unknown.
+                self.jobnames.append(self._get_jobnames(job, testtype, job_identifier))
+                job_group_symbol = job['job_group_symbol'] \
+                    if job['job_group_symbol'] != '?' else ''
+                job_type_symbol = job['job_type_symbol'] if job['job_type_symbol'] else ''
+                self.tbplnames.update({testtype: {'group': job_group_symbol,
```

Comment:
I believe self.tbplnames is only used in the display of the jobs on the seta webpage: http://seta-dev.herokuapp.com/seta.html.  Likewise the entire stack of group_code/group_symbols and related helper functions and data.  We can leave this in, but if we land- I would like to have a bug filed for removing this and just using existing treeherder display tactics.